topic_1/MarketManager.java

```diff
@@ -0,0 +1,36 @@
+package topic_1;
+
+import java.util.ArrayList;
+
+public class MarketManager implements Observer {
```

Comment:
Market Manager must not be modeled as an Object. Notifications are by email. There should be a mailing list that will contain Market Manager's email and the Observer should be a EmailNotifactionService with a mock implementation.